impl/core/src/main/java/io/serverlessworkflow/impl/executors/RunShellExecutor.java

```diff
@@ -0,0 +1,162 @@
+    ProcessBuilder processBuilder = this.processBuilderSupplier.apply(taskContext, workflowContext);
+    WorkflowModel workflowModel =
+        this.shellResultSupplier.apply(taskContext, processBuilder, input);
+    return CompletableFuture.completedFuture(workflowModel);
+  }
+
+  @Override
+  public void init(RunShell taskConfiguration, WorkflowDefinition definition) {
+    Shell shell = taskConfiguration.getShell();
+    final String shellCommand = shell.getCommand();
+
+    this.processBuilderSupplier =
+        (taskContext, workflowContext) -> {
+          WorkflowApplication application = definition.application();
+
+          String command =
+              ExpressionUtils.isExpr(shellCommand)
+                  ? WorkflowUtils.buildStringResolver(
+                          application, shellCommand, taskContext.input().asJavaObject())
+                      .apply(workflowContext, taskContext, taskContext.input())
+                  : shellCommand;
+
+          // TODO: support Windows cmd.exe
+          ProcessBuilder builder = new ProcessBuilder("sh", "-c", command);
+
+          if (shell.getEnvironment() != null
+              && shell.getEnvironment().getAdditionalProperties() != null) {
+            for (Map.Entry<String, Object> entry :
+                shell.getEnvironment().getAdditionalProperties().entrySet()) {
+              String value =
+                  ExpressionUtils.isExpr(entry.getValue())
+                      ? WorkflowUtils.buildStringResolver(
+                              application,
+                              entry.getValue().toString(),
+                              taskContext.input().asJavaObject())
+                          .apply(workflowContext, taskContext, taskContext.input())
+                      : entry.getValue().toString();
+              builder.environment().put(entry.getKey(), value);
+            }
+          }
+
+          return builder;
+        };
+
+    this.shellResultSupplier =
+        (taskContext, processBuilder, input) -> {
+          if (shellCommand == null || shellCommand.isBlank()) {
+            throw new IllegalStateException(
+                "Missing shell command in RunShell task: " + taskContext.taskName());
+          }
+
+          try {
+            Process process = processBuilder.start();
+
+            if (taskConfiguration.isAwait()) {
+              return executeAwaiting(taskConfiguration, definition, process);
+            } else {
+              return input;
+            }
+
+          } catch (IOException | InterruptedException e) {
+            throw new RuntimeException(e);
+          }
+        };
+  }
+
+  private WorkflowModel executeAwaiting(
+      RunShell taskConfiguration, WorkflowDefinition definition, Process process)
+      throws IOException, InterruptedException {
+    StringBuilder stdout = new StringBuilder();
+    StringBuilder stderr = new StringBuilder();
```

Comment:
Althour the override of outputstream is quite imaginative and I like it, I have to mention that is better to use standard library, using
```

StringWriter outWriter = new StringWriter();
process.getInputStream().transferTo(new OutputStreamWriter(outWriter));

.....

outWriter.toString();
```